src/box/xrow.c

```diff
@@ -142,16 +142,23 @@ dump_row_hex(const char *start, const char *end) {
 	}
 }
 
-/**
- * Sets diag and dumps the row body if present.
- */
-#define xrow_on_decode_err(row, what, desc_str) do {\
-	diag_set(ClientError, what, desc_str);\
-	if (row->bodycnt > 0) {\
-		dump_row_hex(row->body[0].iov_base,\
-			     row->body[0].iov_base + row->body[0].iov_len);\
-	}\
-} while (0)
+/** Log corrupted row. */
+static inline void
+xrow_dump_row(const struct xrow_header *row)
+{
+	if (row->bodycnt > 0) {
+		dump_row_hex(row->body[0].iov_base,
+			     row->body[0].iov_base + row->body[0].iov_len);
+	}
+}
+
+/** Set diag and log corrupted row. */
+static inline void
+xrow_on_decode_err(const struct xrow_header *row, int code, const char *desc)
+{
+	diag_set(ClientError, code, desc);
```

Comment:
This must be a macro - for the error file/line. Please revert. Better convert `ER_ILLEGAL_PARAMS` to `IllegalParams` right in the `xrow_on_decode_err` macro.

Reply:
Done. New diff for `xrow.c`:

```diff
 #define xrow_on_decode_err(row, what, desc_str) do {\
-       diag_set(ClientError, what, desc_str);\
+       if (what == ER_ILLEGAL_PARAMS) \
+               diag_set(IllegalParams, desc_str);\
+       else \
+               diag_set(ClientError, what, desc_str);\
        if (row->bodycnt > 0) {\
```